### PR TITLE
[XNIO-378] Improve logging on protocol/port mismatch

### PR DESCRIPTION
**Pull request by spyrkob**
Issue: https://issues.redhat.com/browse/XNIO-378, https://issues.redhat.com/browse/EJBCLIENT-380
3.7 PR: #231 
3.8 PR: #237

### CODE REVIEW

**Comment by dmlloyd** [comment]
How does this solve the problem?  I feel like this is not the right solution but it's not clear how this is supposed to solve EJBCLIENT-380.

**Comment by spyrkob** [comment]
@dmlloyd the idea was to give ejb-client a bit more view into failures. It seems a common case to fail at this point due to port/protocol misconfiguration between client and server, so if ejb-client receives ConnectionClosedEarlyException it can suggest user to look there.

**Comment by dmlloyd** [comment]
I understand the purpose, I just don't understand how this change actually achieves the purpose.  Does this exception then appear to the user?  Does the EJB client catch it and do something specific?

**Comment by spyrkob** [comment]
EJB client catches the exception and logs a message like: https://github.com/wildfly/jboss-ejb-client/pull/466

Sorry I forgot to attach this link to previous message.

File: api/src/main/java/org/xnio/http/HttpUpgrade.java [hunk 1]
```diff
@@ -525,4 +526,9 @@ public void handleCancelled(Object attachment) {
             }
         }
     }
+    public static class ConnectionClosedEarlyException extends EOFException {
```
Comment (dmlloyd) [inline api/src/main/java/org/xnio/http/HttpUpgrade.java]: Exception classes should not be nested classes, and they also need to have serialization information, JavaDoc, and appropriate constructors.  I'd make this a top level class and make it extend `UpgradeFailedException` instead.

**Review by dmlloyd — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/xnio/http/HttpUpgrade.java (hunk 1 above)
Comment (spyrkob) [reply api/src/main/java/org/xnio/http/HttpUpgrade.java]: Thanks, updated now. If it looks good I'll update the 3.7 and ejb-client PRs on Monday

**Review by dmlloyd — APPROVED** [review]

**Pull request merged**